--- hdf-java-2.11.0/Makefile.in.orig	2014-02-20 23:01:32.000000000 +0100
+++ hdf-java-2.11.0/Makefile.in	2015-01-30 16:18:41.133802666 +0100
@@ -34,6 +34,7 @@
 #make this relative to the source root...
 LIBDIR=$(JH45INSTALLDIR)/lib
 BINDIR=$(JH45INSTALLDIR)/bin
+JARDIR=$(JH45INSTALLDIR)/share/java
 DOCDIR= $(JH45INSTALLDIR)/docs
 
 CLASS_DIR=$(TOP)/lib
@@ -282,15 +283,12 @@
 
 install-lib: natives
 	-mkdir -p $(LIBDIR);
-	-mkdir -p $(LIBDIR)/@JAVATARG@
 	@if test x'$(H5INC)' != x'' ; then \
-	cp $(CLASS_DIR)/@JAVATARG@/libjhdf5.$(JSLEXT) $(LIBDIR)/@JAVATARG@ ;  \
-	cp $(H5INC)/../lib/*.settings $(LIBDIR)/@JAVATARG@ ; fi
+	cp $(CLASS_DIR)/@JAVATARG@/libjhdf5.$(JSLEXT) $(LIBDIR) ; fi
 	@if test x'$(H4INC)' != x'' ; then \
-	cp $(CLASS_DIR)/@JAVATARG@/libjhdf.$(JSLEXT) $(LIBDIR)/@JAVATARG@ ;  \
-	cp $(H4INC)/../lib/*.settings $(LIBDIR)/@JAVATARG@ ; fi
+	cp $(CLASS_DIR)/@JAVATARG@/libjhdf.$(JSLEXT) $(LIBDIR) ; fi
 	@if test x'$(H45INC)' != x'' ; then \
-	cp $(CLASS_DIR)/@JAVATARG@/libjh4toh5.$(JSLEXT) $(LIBDIR)/@JAVATARG@ ; fi
+	cp $(CLASS_DIR)/@JAVATARG@/libjh4toh5.$(JSLEXT) $(LIBDIR) ; fi
 	@echo "Install Natives complete"
 
 install-hdflib: 
@@ -298,8 +296,7 @@
 
 do-install-hdflib: hdflib
 	-mkdir -p $(LIBDIR);
-	-mkdir -p $(LIBDIR)/@JAVATARG@
-	cp $(CLASS_DIR)/@JAVATARG@/libjhdf.$(JSLEXT) $(LIBDIR)/@JAVATARG@
+	cp $(CLASS_DIR)/@JAVATARG@/libjhdf.$(JSLEXT) $(LIBDIR)
 	@echo "Install Native HDF complete"
 
 install-hdf5lib: hdf5lib
@@ -307,8 +304,7 @@
 
 do-install-hdf5lib: hdf5lib
 	-mkdir -p $(LIBDIR);
-	-mkdir -p $(LIBDIR)/@JAVATARG@
-	cp $(CLASS_DIR)/@JAVATARG@/libjhdf5.$(JSLEXT) $(LIBDIR)/@JAVATARG@
+	cp $(CLASS_DIR)/@JAVATARG@/libjhdf5.$(JSLEXT)
 	echo "Install Native HDF5 complete"
 
 uninstall-lib:
@@ -321,47 +317,46 @@
 	@if test x'$(H4INC)' != x'' ; then $(MAKE) do-install-jhdf ; fi
 
 do-install-jhdf: jhdf-packages
-	-mkdir -p $(LIBDIR);
-	$(JAR) cf $(JH45INSTALLDIR)/lib/jhdf.jar -C $(CLASS_DIR) $(JHICLASSES)
+	-mkdir -p $(JARDIR);
+	$(JAR) cf $(JARDIR)/jhdf.jar -C $(CLASS_DIR) $(JHICLASSES)
 	@echo "Install JHI complete"
 
 install-jhdf5: 
 	@if test x'$(H5INC)' != x'' ; then $(MAKE) do-install-jhdf5 ; fi
 
 do-install-jhdf5: jhdf5-packages
-	-mkdir -p $(LIBDIR);
-	$(JAR) cf $(JH45INSTALLDIR)/lib/jhdf5.jar -C $(CLASS_DIR) $(JHI5CLASSES)
+	-mkdir -p $(JARDIR);
+	$(JAR) cf $(JARDIR)/jhdf5.jar -C $(CLASS_DIR) $(JHI5CLASSES)
 	@echo "Install JHI5 complete"
 
 install-jhdfobj: ncsa.hdf.object
-	-mkdir -p $(LIBDIR);
-	$(JAR) cf $(JH45INSTALLDIR)/lib/jhdfobj.jar -C $(CLASS_DIR) $(HDFOBJCLASSES)
+	-mkdir -p $(JARDIR);
+	$(JAR) cf $(JARDIR)/jhdfobj.jar -C $(CLASS_DIR) $(HDFOBJCLASSES)
 	@if test x'$(H4INC)' != x'' ; then $(MAKE) do-install-jhdf4obj ; fi
 	@if test x'$(H5INC)' != x'' ; then $(MAKE) do-install-jhdf5obj ; fi
 	@echo "Install HDFOBJ complete"
 
 do-install-jhdf4obj: ncsa.hdf.object.h4
-	-mkdir -p $(LIBDIR);
-	$(JAR) cf $(JH45INSTALLDIR)/lib/jhdf4obj.jar -C $(CLASS_DIR) $(H4OBJCLASSES)
+	-mkdir -p $(JARDIR);
+	$(JAR) cf $(JARDIR)/jhdf4obj.jar -C $(CLASS_DIR) $(H4OBJCLASSES)
 	@echo "Install JHDF4 complete"
 
 do-install-jhdf5obj: ncsa.hdf.object.h5
-	-mkdir -p $(LIBDIR);
-	$(JAR) cf $(JH45INSTALLDIR)/lib/jhdf5obj.jar -C $(CLASS_DIR) $(H5OBJCLASSES)
+	-mkdir -p $(JARDIR);
+	$(JAR) cf $(JARDIR)/jhdf5obj.jar -C $(CLASS_DIR) $(H5OBJCLASSES)
 	@echo "Install JHDF4 complete"
 
 install-jhdfview: ncsa.hdf.view
-	-mkdir -p $(LIBDIR);
-	$(JAR) cf $(JH45INSTALLDIR)/lib/jhdfview.jar -C $(CLASS_DIR) $(HDFVIEWCLASSES) $(HDFVIEWICONS)
-	-mkdir -p $(LIBDIR)/ext
-	$(JAR) cf $(JH45INSTALLDIR)/lib/ext/nc2obj.jar -C $(CLASS_DIR) $(NC2OBJCLASSES)
-	$(JAR) cf $(JH45INSTALLDIR)/lib/ext/fitsobj.jar -C $(CLASS_DIR) $(FITSOBJCLASSES)
+	-mkdir -p $(JARDIR);
+	$(JAR) cf $(JARDIR)/jhdfview.jar -C $(CLASS_DIR) $(HDFVIEWCLASSES) $(HDFVIEWICONS)
+	$(JAR) cf $(JARDIR)/nc2obj.jar -C $(CLASS_DIR) $(NC2OBJCLASSES)
+	$(JAR) cf $(JARDIR)/fitsobj.jar -C $(CLASS_DIR) $(FITSOBJCLASSES)
 	-mkdir -p $(BINDIR);
-	cp lib/netcdf.jar $(LIBDIR)
-	cp lib/fits.jar $(LIBDIR)
-	cp lib/junit.jar $(LIBDIR)
-	cp lib/slf4j-api-1.7.5.jar $(LIBDIR)
-	cp lib/ext/slf4j-nop-1.7.5.jar $(LIBDIR)
+	cp lib/netcdf.jar $(JARDIR)
+	cp lib/fits.jar $(JARDIR)
+	cp lib/junit.jar $(JARDIR)
+	cp lib/slf4j-api-1.7.5.jar $(JARDIR)
+	cp lib/ext/slf4j-nop-1.7.5.jar $(JARDIR)
 	cp bin/hdfview.sh $(BINDIR)
 	chmod a+x $(BINDIR)/hdfview.sh
 	@echo "Install HDFVIEW complete"
