--- hdfjava-3.3.2/Makefile.in.orig	2017-07-25 17:55:31.000000000 +0200
+++ hdfjava-3.3.2/Makefile.in	2020-01-29 17:30:05.635925542 +0100
@@ -34,6 +34,7 @@
 #make this relative to the source root...
 LIBDIR=$(JH45INSTALLDIR)/lib
 BINDIR=$(JH45INSTALLDIR)/bin
+JARDIR=$(JH45INSTALLDIR)/share/java
 DOCDIR= $(JH45INSTALLDIR)/docs
 
 CLASS_DIR=$(TOP)/lib
@@ -206,15 +207,12 @@
 
 install-lib: natives
 	-mkdir -p $(LIBDIR);
-	-mkdir -p $(LIBDIR)/@JAVATARG@
 	@if test x'$(H5INC)' != x'' ; then \
-	cp $(CLASS_DIR)/@JAVATARG@/libjhdf5.$(JSLEXT) $(LIBDIR)/@JAVATARG@ ;  \
-	cp $(H5INC)/../lib/*.settings $(LIBDIR)/@JAVATARG@ ; fi
+	cp $(CLASS_DIR)/@JAVATARG@/libjhdf5.$(JSLEXT) $(LIBDIR) ; fi
 	@if test x'$(H4INC)' != x'' ; then \
-	cp $(CLASS_DIR)/@JAVATARG@/libjhdf.$(JSLEXT) $(LIBDIR)/@JAVATARG@ ;  \
-	cp $(H4INC)/../lib/*.settings $(LIBDIR)/@JAVATARG@ ; fi
+	cp $(CLASS_DIR)/@JAVATARG@/libjhdf.$(JSLEXT) $(LIBDIR) ; fi
 	@if test x'$(H45INC)' != x'' ; then \
-	cp $(CLASS_DIR)/@JAVATARG@/libjh4toh5.$(JSLEXT) $(LIBDIR)/@JAVATARG@ ; fi
+	cp $(CLASS_DIR)/@JAVATARG@/libjh4toh5.$(JSLEXT) $(LIBDIR) ; fi
 	@echo "Install Natives complete"
 
 install-hdflib: 
@@ -222,8 +220,7 @@
 
 do-install-hdflib: hdflib
 	-mkdir -p $(LIBDIR);
-	-mkdir -p $(LIBDIR)/@JAVATARG@
-	cp $(CLASS_DIR)/@JAVATARG@/libjhdf.$(JSLEXT) $(LIBDIR)/@JAVATARG@
+	cp $(CLASS_DIR)/@JAVATARG@/libjhdf.$(JSLEXT) $(LIBDIR)
 	@echo "Install Native HDF complete"
 
 install-hdf5lib: hdf5lib
@@ -231,8 +228,7 @@
 
 do-install-hdf5lib: hdf5lib
 	-mkdir -p $(LIBDIR);
-	-mkdir -p $(LIBDIR)/@JAVATARG@
-	cp $(CLASS_DIR)/@JAVATARG@/libjhdf5.$(JSLEXT) $(LIBDIR)/@JAVATARG@
+	cp $(CLASS_DIR)/@JAVATARG@/libjhdf5.$(JSLEXT)
 	echo "Install Native HDF5 complete"
 
 uninstall-lib:
@@ -245,23 +241,23 @@
 	@if test x'$(H4INC)' != x'' ; then $(MAKE) do-install-jhdf ; fi
 
 do-install-jhdf: jhdf-packages
-	-mkdir -p $(LIBDIR);
-	$(JAR) cf $(JH45INSTALLDIR)/lib/jhdf.jar -C $(CLASS_DIR) $(JHICLASSES)
+	-mkdir -p $(JARDIR);
+	$(JAR) cf $(JARDIR)/jhdf.jar -C $(CLASS_DIR) $(JHICLASSES)
 	@echo "Install JHI complete"
 
 install-jhdf5: 
 	@if test x'$(H5INC)' != x'' ; then $(MAKE) do-install-jhdf5 ; fi
 
 do-install-jhdf5: jhdf5-packages
-	-mkdir -p $(LIBDIR);
-	$(JAR) cf $(JH45INSTALLDIR)/lib/jhdf5.jar -C $(CLASS_DIR) $(JHI5CLASSES)
+	-mkdir -p $(JARDIR);
+	$(JAR) cf $(JARDIR)/jhdf5.jar -C $(CLASS_DIR) $(JHI5CLASSES)
 	@echo "Install JHI5 complete"
 
 uninstall-jhi5: 
-	$(RM) -f $(JH45INSTALLDIR)/lib/jhdf5.jar
+	$(RM) -f $(JARDIR)/jhdf5.jar
 
 uninstall-jhi: 
-	$(RM) -f $(JH45INSTALLDIR)/lib/jhdf.jar
+	$(RM) -f $(JARDIR)/jhdf.jar
 
 install-docs: install-javadocs
 
