--- hdfjava-3.3.2/configure.in.orig	2017-07-25 17:55:29.000000000 +0200
+++ hdfjava-3.3.2/configure.in	2020-01-29 17:15:02.134153565 +0100
@@ -114,21 +114,21 @@
 LLEXT="a";
 JSLEXT="so";
 ;;
-x86_64-pc-linux*) LD=$CC; LDOPT=-shared; SLEXT="so";
+x86_64-*-linux*) LD=$CC; LDOPT=-shared; SLEXT="so";
 TARGETLIB="linux"
 SLEXT="so";
 LLEXT="a";  dnl the extension for the HDF and other libraries to link to
 JSLEXT="so";
 COPT="-shared -Dx86_64 -fPIC" ;
 ;;
-ia64-pc-linux*) LD=$CC; LDOPT=-shared; SLEXT="so";
+ia64-*-linux*) LD=$CC; LDOPT=-shared; SLEXT="so";
 TARGETLIB="linux"
 SLEXT="so";
 LLEXT="a";  dnl the extension for the HDF and other libraries to link to
 JSLEXT="so";
 COPT="-shared -Dia64 -fPIC" ;
 ;;
-*-pc-linux*) LD=$CC; LDOPT=-shared; SLEXT="so";
+*-linux*) LD=$CC; LDOPT=-shared; SLEXT="so";
 TARGETLIB="linux"
 SLEXT="so";
 LLEXT="a";  dnl the extension for the HDF and other libraries to link to
@@ -169,8 +169,8 @@
 
 case "$withval" in
   yes)
-      ZLIB=""
-  HAVE_ZLIB="no"
+      ZLIB="-lz"
+  HAVE_ZLIB="yes"
     ;;
   no)
       ZLIB=""
@@ -204,7 +204,8 @@
 
 case "$withval" in
   yes)
-  HAVE_JPEG="no"
+	JPEGLIB="-ljpeg"
+  HAVE_JPEG="yes"
     ;;
   no)
       JPEGLIB=""
@@ -238,8 +239,8 @@
 
 case "$withval" in
   yes)
-      SZLIB=""
-  HAVE_SZIP="no"
+      SZLIB="-lsz"
+  HAVE_SZIP="yes"
     ;;
   no)
       SZLIB=""
@@ -279,8 +280,8 @@
 case "$withval" in
   yes)
     dnl AC_MSG_ERROR(HDF4 library needed for compilation.)
-      HDF4LIB=""
-  HAVE_HDF4="no"
+      HDF4LIB="-ldf -lmfhdf"
+  HAVE_HDF4="yes"
     ;;
   no)
       HDF4LIB=""
@@ -294,7 +295,7 @@
       fi
       hdf4_lib="`echo $withval |cut -f2 -d, -s`"
       if test -n "$hdf4_lib"; then
-    HDF4LIB=$hdf4_lib
+    HDF4LIB="-ldf -lmfhdf"
       fi
       HAVE_HDF4="yes"
   else
@@ -304,7 +305,7 @@
   ;;
 esac
 if test -n "$HDF4LIB"; then
-  LIBS="$HDF4LIB/libdf.$LLEXT $HDF4LIB/libmfhdf.$LLEXT $LIBS"
+  LIBS="$HDF4LIB $LIBS"
   HAVE_HDF4="yes"
 fi
 AC_MSG_RESULT([$HAVE_HDF4])
@@ -355,8 +356,8 @@
 
 case "$withval" in
   yes)
-      HDF5LIB=""
-  HAVE_HDF5="no"
+      HDF5LIB="-lhdf5"
+  HAVE_HDF5="yes"
     ;;
   no)
       HDF5LIB=""
@@ -364,15 +365,13 @@
     ;;
   *)
   if test -n "$withval" ; then
-      hdf5_lib="`echo $withval |cut -f2 -d, -s`"
-    HDF5LIB=$hdf5_lib"/libhdf.$LLEXT"
       hdf5_inc="`echo $withval |cut -f1 -d,`"
       if test -n "$hdf5_inc"; then
         HDF5INC="$hdf5_inc"
       fi
       hdf5_lib="`echo $withval |cut -f2 -d, -s`"
       if test -n "$hdf5_lib"; then
-    HDF5LIB=$hdf5_lib
+    HDF5LIB="-lhdf5"
       fi
       HAVE_HDF5="yes"
   else
@@ -382,7 +381,7 @@
   ;;
 esac
 if test -n "$HDF5LIB"; then
-  LIBS="$HDF5LIB/libhdf5.$LLEXT $LIBS"
+  LIBS="$HDF5LIB $LIBS"
   HAVE_HDF5="yes"
 fi
 AC_MSG_RESULT([$HAVE_HDF5])
@@ -397,8 +396,14 @@
 dnl
 if test -n "$HDF5LIB"; then
   AC_MSG_CHECKING([dependencies for HDF5 library])
+  if test "${HDF5LIB#/}" = "$HDF5LIB" ; then
+	  eval eval ldir=$libdir
+  DFL=`grep 'filters (external)' $ldir"/libhdf5.settings" | grep deflate`
+  SZP=`grep 'filters (external)' $ldir"/libhdf5.settings" | grep szip`
+  else
   DFL=`grep 'filters (external)' $HDF5LIB"/libhdf5.settings" | grep deflate`
   SZP=`grep 'filters (external)' $HDF5LIB"/libhdf5.settings" | grep szip`
+  fi
   if test -n "$DFL";
   then
     if test "$HAVE_ZLIB" != "yes"; then
@@ -613,7 +618,7 @@
 JSLEXT="so";
 JAVAINC2=$JAVAINC/irix; JAVATARG=irix-$OS;
 ;;
-x86_64-pc-linux*) LD=$CC; LDOPT=-shared;
+x86_64-*-linux*) LD=$CC; LDOPT=-shared;
 SLEXT="so";
 LLEXT="a";  # the extension for the HDF and other libraries to link to
 JSLEXT="so";
@@ -621,7 +626,7 @@
 JAVAINC2=$JAVAINC/linux;
 JAVATARG=linux;
 ;;
-ia64-pc-linux*) LD=$CC; LDOPT=-shared;
+ia64-*-linux*) LD=$CC; LDOPT=-shared;
 SLEXT="so";
 LLEXT="a";  # the extension for the HDF and other libraries to link to
 JSLEXT="so";
@@ -629,7 +634,7 @@
 JAVAINC2=$JAVAINC/linux;
 JAVATARG=linux;
 ;;
-*-pc-linux*) LD=$CC; LDOPT=-shared;
+*-linux*) LD=$CC; LDOPT=-shared;
 SLEXT="so";
 LLEXT="a";  # the extension for the HDF and other libraries to link to
 JSLEXT="so";
--- hdf-java/native/hdflib/Makefile.in.orig	2012-11-13 00:11:48.000000000 +0100
+++ hdf-java/native/hdflib/Makefile.in	2013-04-06 07:51:52.942956301 +0200
@@ -41,7 +41,7 @@
 CFLAGS = $(DEFS) -I. -I$(JAVAINC1) -I$(JAVAINC2) -I$(HDFINC)
 
 # Be sure to add -lbsd if compiling in a SYSV environment
-LIBS =  $(HDFLIB)/libmfhdf.$(LLEXT) $(HDFLIB)/libdf.$(LLEXT) $(SZLIB) $(ZLIB) $(JPEGLIB) -lm
+LIBS =  $(HDFLIB) $(SZLIB) $(ZLIB) $(JPEGLIB) -lm
 
 # Macro def: object filename used
 OBJECTS =  hdfstructsutil.o \
--- hdf-java/native/hdf5lib/Makefile.in.orig	2012-11-13 00:11:49.000000000 +0100
+++ hdf-java/native/hdf5lib/Makefile.in	2013-04-06 07:53:18.476287850 +0200
@@ -35,7 +35,7 @@
 
 CFLAGS = -D_FILE_OFFSET_BITS=64 $(DEFS) -I. -I$(HDF5INC) -I$(JAVAINC1) -I$(JAVAINC2)
 
-LIBS =  $(HDF5LIB)/libhdf5.$(LLEXT) $(ZLIB) $(SZLIB) -lm
+LIBS =  $(HDF5LIB) $(ZLIB) $(SZLIB) -lm
 
 # Macro def: object filename used
 OBJECTS =  exceptionImp.o h5Constants.o h5Imp.o h5aImp.o h5dImp.o h5eImp.o h5fImp.o h5gImp.o h5iImp.o h5lImp.o h5oImp.o h5pImp.o h5rImp.o h5sImp.o h5tImp.o nativeData.o h5util.o h5zImp.o
--- hdfjava-3.3.2/native/hdf5lib/Makefile.in.orig	2020-02-02 18:31:06.573816725 +0100
+++ hdfjava-3.3.2/native/hdf5lib/Makefile.in	2020-02-03 17:32:43.558058582 +0100
@@ -32,6 +32,7 @@
 JSLEXT=@JSLEXT@
 
 COPT=@COPT@
+CPPFLAGS=@CPPFLAGS@
 
 CFLAGS = -D_FILE_OFFSET_BITS=64 $(DEFS) -I. -I$(HDF5INC) -I$(JAVAINC1) -I$(JAVAINC2)
 
